### PR TITLE
Gettexed - almost all the work to multilingual app

### PR DESCRIPTION
**Pull request by enedil**
Hi!
What I made?
I integrated Gettext. Some of the strings need to be changed to form  `_("string")`.
And finally, someone should translate it. I included a markdown file with instructions.
However, this change does not break Pyromaths and thus I think it is good.

### CODE REVIEW

File: src/pyromaths/ex/cinquiemes/aires.py [hunk 1]
```diff
@@ -40,13 +40,13 @@ def exo_aire_diques():
...
-        donnees = ('rayons', rayon1, rayon2)
+        donnees = (_('rayons'), rayon1, rayon2)
     else:
-        donnees = (u'diamètres', 2 * rayon1, 2 * rayon2)
+        donnees = (_(u'diamètres'), 2 * rayon1, 2 * rayon2)
     difference_des_carres = rayon2 ** 2 - rayon1 ** 2
     aire_arrondie = arrondir(str(3.14 * difference_des_carres))
     enonce = \
-        u"""\\begin{minipage}{4cm}
+        _(u"""\\begin{minipage}{4cm}
```
Comment (JeromeO) [inline src/pyromaths/ex/cinquiemes/aires.py]: Useless, this is a TeX command.

File: src/pyromaths/ex/cinquiemes/aires.py [hunk 2]
```diff
@@ -56,28 +56,28 @@ def exo_aire_diques():
 \\end{minipage}\\hfill
 \\begin{minipage}{13cm}
 On considère deux cercles de centre $O$ et de %s respectifs $\\unit[%s]{cm}$ et $\\unit[%s]{cm}$.\\par
-Calculer l'aire de la couronne circulaire (partie colorée) comprise entre les deux cercles en arrondissant le résultat au $\\unit{cm^2}$ le plus proche.""" % donnees
+Calculer l'aire de la couronne circulaire (partie colorée) comprise entre les deux cercles en arrondissant le résultat au $\\unit{cm^2}$ le plus proche.""") % donnees
     exo.append(enonce)
     cor.append(enonce)
-    cor.append("\\par\\dotfill{}\\\\\n")
+    cor.append(_("\\par\\dotfill{}\\\\\n"))
```
Comment (JeromeO) [inline src/pyromaths/ex/cinquiemes/aires.py]: Useless, this is a TeX command.

File: src/pyromaths/ex/cinquiemes/statistiques.py [hunk 3]
```diff
@@ -76,46 +76,46 @@ def tableau_tex(titres, larg="c", eff=1, freq=1, val=[[], []], total=1):
...
     return tableau_tex
 
 def diagramme_tex(typed=2, val=[[], []], aide=0):
     """Génère un diagramme en bâtons (type 1), circulaire (type2) ou semi-circulaire (type3) à partir des fréquences (liste val[1]) et son tableau de calculs."""
     diag = ""
-    
-    couleur = ["AliceBlue", "Bisque", "PaleGreen", "Thistle", "LightGray", "Khaki", "LightBlue", "LightSalmon", "PaleGoldenrod", "PapayaWhip", "Plum", "Gainsboro"]
-    
+
+    couleur = [_("AliceBlue"), _("Bisque"), _("PaleGreen"), _("Thistle"), _("LightGray"), _("Khaki"), _("LightBlue"), _("LightSalmon"), _("PaleGoldenrod"), _("PapayaWhip"), _("Plum"), _("Gainsboro")]
```
Comment (JeromeO) [inline src/pyromaths/ex/cinquiemes/statistiques.py]: Useless, this is a TeX command.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
 [Pyromaths](http://pyromaths.org/) est un programme qui a pour but de créer des exercices type de mathématiques niveau collège et lycée ainsi que leur corrigé. C'est ce qu'on appelle parfois un exerciseur. Contrairement à de nombreux autres projets, Pyromaths a pour objectif de proposer une correction véritablement détaillée des exercices proposés et pas seulement une solution.
 
 Il permet par exemple de proposer des devoirs maison aux élèves et de leur distribuer ensuite la correction. Il peut aussi servir à des familles afin qu'un élève puisse travailler un point du programme et se corriger ensuite.
+
+Si vous voulez aider avec la traduction, consultez [cette page](https://github.com/Pyromaths/pyromaths/blob/develop/locale/TRADUIRE.md).
```
Comment (JeromeO) [inline README.md]: Si vous voulez participer à la traduction, …

File: locale/TRADUIRE.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+### Windows:
+Gettext est [là](http://gnuwin32.sourceforge.net/packages/gettext.htm).
+
+
+## Géneration des fichiers à traduire
+Pour obtenir le fichier `.po`, lancez
+```
+xgettext -L Python -d pyromaths --from-code utf-8 -o locale/pyromaths.pot -f locale/traductibles
+```
+depuis directoire racine du code source.
```
Comment (JeromeO) [inline locale/TRADUIRE.md]: It’s not « directoire » but « répertoire »

**Comment by JeromeO** [comment]
Really nice work. I made some comments.

File: README.md (hunk 4 above)
Comment (enedil) [inline README.md]: I really appreciate your help. All my french was at school. I do not quite understand the semantics well.

File: src/pyromaths/ex/cinquiemes/aires.py (hunk 1 above)
Comment (enedil) [inline src/pyromaths/ex/cinquiemes/aires.py]: Even the u'diamètres'?
You definied something, right?
I was in a hurry. I these .tex commands - sometimes I did not think about them and just made them wrapped.
I'll add a commit which unbreakes the tex strings.

**Comment by enedil** [comment]
What now? Are you going to merge it?

**Comment by JeromeO** [comment]
I will when I'll be back from holidays. 
---
Envoyé de mon téléphone mobile. 

enedil <notifications@github.com> a écrit :

**Comment by enedil** [comment]
@JeromeO that's fine. Thanks for the info.

**Comment by enedil** [comment]
@JeromeO 
It is still in french, because nobody bothered to add the actual translations.
If there is a volunteer who translates the strings from french to english, there will be english application.
You could try to clone my repo locally, add changes made by @Tarzan1984 (just move one directory to the other - it will merge) and run it as

    LANG="pl_PL.UTF-8" LC_MESSAGES=pl python src/pyromaths.py

File: src/pyromaths/ex/cinquiemes/aires.py (hunk 1 above)
Comment (paternal) [inline src/pyromaths/ex/cinquiemes/aires.py]: Maybe this string should be splitted: the beginning (containing all the LaTeX commands) should be less as is, while the last part (containing the French text) should be translated.

**Pull request merged**